### ISSUE TITLE
Error while migrating code block

### ISSUE THREAD

**Issue by MasGaNo** [TryGhost/migrate#806]
Hello the team,

I'm facing an error while using the migrate tools to convert medium.com posts to Ghost posts:
```sh
migrate medium --pathToZip medium-export-test.zip
```

Medium.com code block:
<img width="703" alt="image" src="https://github.com/TryGhost/migrate/assets/5469433/9ff794ba-6462-41ad-99b4-d2feabda9460">

Actual Ghost.io code block:
<img width="734" alt="image" src="https://github.com/TryGhost/migrate/assets/5469433/d9170b1c-c9ef-4793-b4ed-31eccc3d7f1c">

Expected Ghost.io code block:
<img width="728" alt="image" src="https://github.com/TryGhost/migrate/assets/5469433/7fcad0d2-6e58-4727-bbff-f6187b73e656">

Medium.com exported HTML code:
```html
<pre name="4a0a" id="4a0a" class="graf graf--pre graf-after--p">
  <code class="markup--code markup--pre-code">
    sudo apt-get update<br>sudo apt-get install \<br>    apt-transport-https \<br>    ca-certificates \<br>    curl \<br>    gnupg-agent \<br>    software-properties-common \<br>    certbot \<br>    python3-certbot-nginx
  </code>
</pre>
<pre name="ebeb" id="ebeb" class="graf graf--pre graf-after--pre">
  <code class="markup--code markup--pre-code">
    sudo apt-key adv --keyserver keys.gnupg.net --recv-keys 0xDE8B853FC155581D
  </code>
</pre>
<pre name="9a1f" id="9a1f" class="graf graf--pre graf-after--pre">
  <code class="markup--code markup--pre-code">
    echo &quot;deb https://download.passbolt.com/ce/debian buster stable&quot; |\ <br>sudo tee /etc/apt/sources.list.d/passbolt.list<br>sudo apt-get update<br>sudo apt-get install passbolt-ce-server
  </code>
</pre>
```

Ghost.io mobiledoc:
```js
[
  "code",
  {
    "code": "sudo apt-get updatesudo apt-get install \\    apt-transport-https \\    ca-certificates \\    curl \\    gnupg-agent \\    software-properties-common \\    certbot \\    python3-certbot-nginx"
  }
],
[
  "code",
  {
    "code": "sudo apt-key adv --keyserver keys.gnupg.net --recv-keys 0xDE8B853FC155581D"
  }
],
[
  "code",
  {
    "code": "echo \"deb https://download.passbolt.com/ce/debian buster stable\" |\\ sudo tee /etc/apt/sources.list.d/passbolt.listsudo apt-get updatesudo apt-get install passbolt-ce-server"
  }
]
```

As you can see, all `<br>` tag were stripped during the process of migration. 

It may happen somewhere in the [convertPost](https://github.com/TryGhost/migrate/blob/main/packages/mg-html-mobiledoc/lib/convertPost.js#L19) function which uses `@tryghost/html-to-mobiledoc` under the hood, and itself is using [@tryghost/kg-parser-plugins](https://github.com/TryGhost/Koenig/blob/main/packages/kg-clean-basic-html/lib/clean-basic-html.js#L32) to parse the HTML by removing all `br` tag as the allowBr option is not correctly apply.

I'm using:
```sh
% migrate --version
0.31.1
```

Thank you.

**Comment by MasGaNo** [comment]
Another example:

Medium.com: 
<img width="697" alt="image" src="https://github.com/TryGhost/migrate/assets/5469433/8b476412-a3ac-4cb1-a202-9a3a36cc4296">

Ghost.io:
<img width="736" alt="image" src="https://github.com/TryGhost/migrate/assets/5469433/02ae9b7d-0604-475e-842c-f32b9f1cb75e">

Medium.com HTML:
```html
<p name="4722" id="4722" class="graf graf--p graf-after--h3">
  Well, now that’s out of the way, let’s get started! The first step is to add our <a href="https://github.com/..." data-href="https://github.com/..." class="markup--anchor markup--p-anchor" rel="noopener" target="_blank">repo for the Helm chart</a>:
</p>
<pre data-code-block-mode="2" spellcheck="false" data-code-block-lang="bash" name="2296" id="2296" class="graf graf--pre graf-after--p graf--preV2">
  <span class="pre--content">helm repo add passbolt-repo https://download.passbolt.com/charts/passbolt</span>
</pre>
```

Ghost.io mobiledoc:
```js
[
  1,
  "p",
  [
    [
      0,
      [],
      0,
      "Well, now that’s out of the way, let’s get started! The first step is to add our "
    ],
    [
      0,
      [6],
      1,
      "repo for the Helm chart"
    ],
    [
      0,
      [],
      0,
      ":helm repo add passbolt-repo https://download.passbolt.com/charts/passbolt"
    ]
  ]
]
```

Here the whole `<pre>` tag was stripped and merge with the previous `p` tag

**Comment by PaulAdamDavis** [comment]
Hi @MasGaNo, thanks for reporting this and providing the source HTML! I'm working on a fix and will update this issue when that's done.

**Comment by PaulAdamDavis** [comment]
This should now be fixed and released in `@tryghost/migrate@0.32.0`. Thanks again for reporting!
